File: app/src/main/java/com/diploma/bank/MainMenu/MainHomeListFragment.java
package com.diploma.bank.MainMenu;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.support.v7.widget.RecyclerView;
import android.content.Intent;

import com.diploma.bank.Authorization.LoginActivity;
import com.diploma.bank.CardAccountActivity;
import com.diploma.bank.R;
import com.diploma.bank.TempData.CardAccounts;
import com.diploma.bank.TempData.Deposits;

public class MainHomeListFragment extends Fragment {

    public MainHomeListFragment() {
    }

    public static MainHomeListFragment newInstance() {
        return new MainHomeListFragment();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        RecyclerView card_account_Recycler = (RecyclerView)inflater.inflate(
                R.layout.fragment_main_home_list, container, false);

        //Список карточных счетов
        final int[] id_card = new int[CardAccounts.account.length];
        for(int i = 0; i < id_card.length; i++){
            id_card[i] = CardAccounts.account[i].getId_card();
        }

        String[] card_names = new String[CardAccounts.account.length];
        for (int i = 0; i < card_names.length; i++) {
            card_names[i] = CardAccounts.account[i].getName_card();
        }

        double[] card_balance = new double[CardAccounts.account.length];
        for (int i = 0; i < card_balance.length; i++) {
            card_balance[i] = CardAccounts.account[i].getBalance();
        }

        int[] ic_images = new int[CardAccounts.account.length];
        for (int i = 0; i < ic_images.length; i++) {
            ic_images[i] = CardAccounts.account[i].getId_image();
        }

        //Список вкладов
        int[] id_deposit = new int[Deposits.deposits.length];
        for(int i = 0; i < id_deposit.length; i++){
            id_deposit[i] = Deposits.deposits[i].getId_deposit();
        }

        String[] deposit_name = new String[Deposits.deposits.length];
        for (int i = 0; i < deposit_name.length; i++) {
            deposit_name[i] = Deposits.deposits[i].getDeposit_name();
        }

        double[] deposit_balance = new double[Deposits.deposits.length];
        for (int i = 0; i < deposit_balance.length; i++) {
            deposit_balance[i] = Deposits.deposits[i].getDeposit_balance();
        }

        int[] deposit_id_image = new int[Deposits.deposits.length];
        for (int i = 0; i < deposit_id_image.length; i++) {
            deposit_id_image[i] = Deposits.deposits[i].getDeposit_id_image();
        }

        MainHomeListAdapter adapter_main_home_list = new MainHomeListAdapter(id_card, card_names, card_balance, ic_images,
                id_deposit, deposit_name, deposit_balance, deposit_id_image);
        card_account_Recycler.setAdapter(adapter_main_home_list);
        LinearLayoutManager layoutManager_main_home_list = new LinearLayoutManager(getActivity());
        card_account_Recycler.setLayoutManager(layoutManager_main_home_list);

        adapter_main_home_list.setListener(new MainHomeListAdapter.Listener(){
            public void onClick(int position) {
                if (position > id_card.length){
                    Intent intent = new Intent(getActivity(), LoginActivity.class);
                    getActivity().startActivity(intent);
                }
                else {
                    Intent intent = new Intent(getActivity(), CardAccountActivity.class);
                    intent.putExtra(CardAccountActivity.EXTRA_BANK_ACC_ID, position);
                    getActivity().startActivity(intent);
                }
            }
        });

        return card_account_Recycler;
    }
}

File: app/src/main/java/com/vmbeshenov/bank/MainMenu/MainHomeListFragment.kt
package com.vmbeshenov.bank.MainMenu

import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.vmbeshenov.bank.Authorization.LoginActivity
import com.vmbeshenov.bank.CardAccount.CardAccountActivity
import com.vmbeshenov.bank.R
import com.vmbeshenov.bank.TempData.CardAccounts
import com.vmbeshenov.bank.TempData.Deposits

class MainHomeListFragment : Fragment() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val cardAccountRecycler = inflater.inflate(
                R.layout.fragment_main_home_list, container, false) as RecyclerView

        //Список карточных счетов
        val idCard = IntArray(CardAccounts.account.size)
        for (i in idCard.indices) {
            idCard[i] = CardAccounts.account[i].id_card
        }
        val cardNames = arrayOfNulls<String>(CardAccounts.account.size)
        for (i in cardNames.indices) {
            cardNames[i] = CardAccounts.account[i].name_card
        }
        val cardBalance = DoubleArray(CardAccounts.account.size)
        for (i in cardBalance.indices) {
            cardBalance[i] = CardAccounts.account[i].balance
        }
        val icImages = IntArray(CardAccounts.account.size)
        for (i in icImages.indices) {
            icImages[i] = CardAccounts.account[i].id_image
        }

        //Список вкладов
        val idDeposit = IntArray(Deposits.deposits.size)
        for (i in idDeposit.indices) {
            idDeposit[i] = Deposits.deposits[i].id_deposit
        }
        val depositName = arrayOfNulls<String>(Deposits.deposits.size)
        for (i in depositName.indices) {
            depositName[i] = Deposits.deposits[i].deposit_name
        }
        val depositBalance = DoubleArray(Deposits.deposits.size)
        for (i in depositBalance.indices) {
            depositBalance[i] = Deposits.deposits[i].deposit_balance
        }
        val depositIdImage = IntArray(Deposits.deposits.size)
        for (i in depositIdImage.indices) {
            depositIdImage[i] = Deposits.deposits[i].deposit_id_image
        }
        val adapterMainHomeList = MainHomeListAdapter(idCard, cardNames, cardBalance, icImages,
                idDeposit, depositName, depositBalance, depositIdImage)
        cardAccountRecycler.adapter = adapterMainHomeList
        val layoutManagerMainHomeList = LinearLayoutManager(activity)
        cardAccountRecycler.layoutManager = layoutManagerMainHomeList
        adapterMainHomeList.setListener(object : MainHomeListAdapter.Listener {
            override fun onClick(position: Int) {
                if (position > idCard.size) {
                    val intent = Intent(activity, LoginActivity::class.java)
                    activity!!.startActivity(intent)
                } else {
                    val intent = Intent(activity, CardAccountActivity::class.java)
                    intent.putExtra(CardAccountActivity.EXTRA_BANK_ACC_ID, position)
                    activity!!.startActivity(intent)
                }
            }
        })
        return cardAccountRecycler
    }

    companion object {
        fun newInstance(): MainHomeListFragment {
            return MainHomeListFragment()
        }
    }
}